src/components/delegateRegistration/selectName/SelectName.js

```diff
@@ -0,0 +1,189 @@
+import React from 'react';
+import debounce from 'lodash.debounce';
+import Box from '../../boxV2';
+import { InputV2 } from '../../toolbox/inputsV2';
+import { PrimaryButtonV2 } from '../../toolbox/buttons/button';
+import Feedback from '../../toolbox/feedback/feedback';
+import SpinnerV2 from '../../spinnerV2/spinnerV2';
+import Icon from '../../toolbox/icon';
+import { fromRawLsk } from '../../../utils/lsk';
+import regex from '../../../utils/regex';
+import links from '../../../constants/externalLinks';
+import Fees from '../../../constants/fees';
+import styles from './selectName.css';
+
+class SelectName extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      nickname: '',
+      error: false,
+      inputDisabled: false,
+      loading: false,
+    };
+
+    this.onChangeNickname = this.onChangeNickname.bind(this);
+    this.debounceFetchUser = debounce(this.onUserFetch, 1000);
+  }
+
+  componentDidMount() {
+    this.getNicknameFromPrevState();
+    this.checkIfUserIsDelegate();
+    this.hasUserHasEnoughFunds();
+  }
+
+  componentDidUpdate(prevProps) {
+    const { delegate, t } = this.props;
+
+    if (prevProps.delegate !== delegate && delegate) {
+      if (delegate.delegateNameInvalid) {
+        this.setState({ loading: false, error: t('Name is already taken!') });
+      } else {
+        this.setState({ loading: false });
+      }
+    }
+  }
+
+  getNicknameFromPrevState() {
+    const { prevState } = this.props;
+    if (Object.entries(prevState).length) this.setState({ nickname: prevState.nickname });
+  }
+
+  hasUserHasEnoughFunds() {
+    const { account, t } = this.props;
+    const hasFunds = account.info &&
+      fromRawLsk(account.info.LSK.balance) * 1 >= fromRawLsk(Fees.registerDelegate) * 1;
+
+    if (!hasFunds) {
+      this.setState({
+        inputDisabled: true,
+        error: t('Insufficient funds (Fee: {{fee}} LSK)', { fee: fromRawLsk(Fees.registerDelegate) }),
+      });
+    }
+  }
+
+  checkIfUserIsDelegate() {
+    const { account, t } = this.props;
+    if (account && account.isDelegate) {
+      this.setState({
+        inputDisabled: true,
+        error: t('You have already registered as a delegate.'),
+      });
+    }
+  }
+
+  isNameInvalid(nickname) {
+    const { t } = this.props;
+    if (nickname.length > 20) return t('Nickname is too long.');
+    const hasInvalidChars = nickname.replace(regex.delegateSpecialChars, '');
+    if (hasInvalidChars) return t(`Invalid character ${hasInvalidChars.trim()}`);
+    return false;
+  }
+
+  onUserFetch(nickname, error) {
+    if (!error && nickname.length) {
+      this.props.delegatesFetched({ username: nickname });
```

Comment:
I think this is one of the cases where we don't need to use redux and can have the API request directly in the component because the data from the API is not used by any other component.

Reply:
yes this is going to be addressed in PR https://github.com/LiskHQ/lisk-hub/pull/2186